memorystore/valkey/caching/GUIDE.md

```diff
@@ -0,0 +1,189 @@
+# Building a Caching Service on Google Cloud using Valkey, Spring Boot, and PostgreSQL
+
+Modern applications need to deliver fast, responsive user experiences at scale.
+
+In this guide, there are architectural concepts and deployment steps for creating a high-performance caching service on Google Cloud. Using a combination of Java, Spring Boot, PostgreSQL, and Valkey, you can reduce latency while also reducing the load on your database.
+
+## Why Caching Matters
+
+- **Speed & Latency:** Storing frequently requested data in memory avoids repeated round-trip queries to databases, while reducing response times.
+- **Scalability:** By reducing the workload on your database, applications can serve data directly from memory, increasing the capacity for requests.
+
+## What You’ll Build
+
+You’ll set up a caching service that:
+
+1. **Works with a PostgreSQL database** to store long-lived, persistent records.
+2. **Incorporates Valkey** as a high-speed, in-memory cache, fronting the PostgreSQL database.
+3. **Uses Spring Boot** to expose REST endpoints, providing a simple interface for reading, writing, and invalidating cached data.
+4. **A solution that can be deployed to Google Cloud Platform (GCP)** for production, leveraging services like Cloud Run, Cloud SQL, and Memorystore.
+
+By following this guide, you’ll have a reference architecture ready to adapt, test, and deploy to meet the performance needs of your application.
+
+## Architecture Overview
+
+- **Spring Boot Application:** Serves as the middle tier for responding to API calls. When a request is received, the API checks Valkey for cached results; if no entries are found, then the API will retrieve data from the PostgreSQL database and update the cache.
+- **Valkey (In-Memory Cache):** A Redis-like memory store that keeps hot data ready to be served instantly.
+- **PostgreSQL Database:** Your source of truth for all data. The cache reduces how often the app queries this database.
+- **Google Cloud Infrastructure:** Deployed using Terraform, you can host the application on Cloud Run, store data in Cloud SQL for PostgreSQL, and leverage Memorystore for Valkey.
+
+## Step-by-step Guide
+
+To begin, we are will generate an API with the following routes:
+
+_create_: For creating new items.
+_delete_: For creating deleting items.
+_create_: For creating new items.
```

Comment:
should it be create,read, update, delete?
or create,read, delete?

same question for following files.